deltaspike/modules/data/impl/src/main/java/org/apache/deltaspike/data/impl/builder/AnnotatedQueryBuilder.java

```diff
@@ -88,4 +83,12 @@ else if (query.isNative())
         return context.applyRestrictions(result);
     }
 
+    private Class<?> getMethodResultEntityClass(Method m){
```

Comment:
[QueryBuilder](https://github.com/apache/deltaspike/blob/master/deltaspike/modules/data/impl/src/main/java/org/apache/deltaspike/data/impl/builder/QueryBuilder.java#L64) also only checks for java.util.List